### PR TITLE
Fix a few small issues in service catalog uninstall

### PR DESCRIPTION
**Pull request by staebler**
* Service catalog install was not re-creating the apiserver.crt and apiserver.key files when generating certs. But the ca.crt and ca.key files were being re-created. This was causing non-verifiable certs to be used when service catalog was uninstalled and re-installed. The service catalog installer was changed to delete the apiserver.crt and apiserver.key files so that they are re-created using the new ca.crt and ca.key files.

* The asb auth token secret was not being deleted correctly and causing the uninstaller to fail.

* The asb uninstaller was attempting to delete the broker registration from the service catalog. However, the service catalog is uninstalled first. When the asb uninstaller would fail when attempting to delete the ClusterServiceBroker. The uninstaller was changed to verify that the servicecatalog APIService exists first before attempting to delete the ClusterServiceBroker.

* The service catalog uninstaller was attempting to delete policybindings. The server does not have a resource type name policybinding. I do not know what the intention is there, but I have commented out that part of the uninstaller.

### CODE REVIEW

File: roles/openshift_service_catalog/tasks/remove.yml [hunk 1]
```diff
@@ -3,9 +3,10 @@
   command: >
     oc delete apiservices.apiregistration.k8s.io/v1alpha1.servicecatalog.k8s.io --ignore-not-found -n kube-service-catalog
 
-- name: Remove Policy Binding
-  command: >
-    oc delete policybindings/kube-system:default -n kube-system --ignore-not-found
+# TODO: policybinding is not a resource type. what was the original intention of this?
```
Comment (ewolinetz) [inline roles/openshift_service_catalog/tasks/remove.yml]: We could just remove this if it wouldn't actually do anything otherwise.

Comment (staebler) [reply roles/openshift_service_catalog/tasks/remove.yml]: We can remove it. My concern is that there may be something that should be done here. If we remove it, then we lose the marker that reminds us that something needs to be done. I haven't looked into it more deeply to understand what if anything needs to be done here.

Comment (ewolinetz) [reply roles/openshift_service_catalog/tasks/remove.yml]: @deads2k @bparees do you happen to know if there is a concern with this hanging around after a removal?

Comment (deads2k) [reply roles/openshift_service_catalog/tasks/remove.yml]: > @deads2k @bparees do you happen to know if there is a concern with this hanging around after a removal?

It won't exist at all in 3.7.  There's nothing to delete.

Comment (staebler) [reply roles/openshift_service_catalog/tasks/remove.yml]: That was my point. There is nothing to delete, but this was deleting something at some point. Presumably whatever it was deleting has been replaced by something else. What would it have been replaced with? And is there anything we need to do here to delete the replacement.

Comment (deads2k) [reply roles/openshift_service_catalog/tasks/remove.yml]: > That was my point. There is nothing to delete, but this was deleting something at some point. Presumably whatever it was deleting has been replaced by something else. What would it have been replaced with? And is there anything we need to do here to delete the replacement.

It was not replaced with anything.  It did not have value.

Comment (deads2k) [reply roles/openshift_service_catalog/tasks/remove.yml]: You are removing your rolebinding.rbac objects, right?  If you were counting on this having the side-effect of removing rolebindings (never recommended), then you have to delete rolebindings.rbac.

Comment (staebler) [reply roles/openshift_service_catalog/tasks/remove.yml]: I have removed the commented-out code that attempted to remove the policybinding. I have replaced it with code to delete the role-bindings via the templates created during install.

File: roles/ansible_service_broker/tasks/remove.yml [hunk 2]
```diff
@@ -99,11 +99,17 @@
...
+  register: get_apiservices
+  changed_when: no
+
 - name: remove broker object from the catalog
   oc_obj:
     name: ansible-service-broker
     state: absent
-    kind: ServiceBroker
+    kind: ClusterServiceBroker
+  when: not "'not found' in get_apiservices.stdout"
```
Comment (ewolinetz) [inline roles/ansible_service_broker/tasks/remove.yml]: i'm surprised this isn't being flagged... i would think this should be
`"not 'not found' in get_apiservices.stdout"`

**Comment by ewolinetz** [comment]
hey bot, why not merge this?

**Pull request merged**